src/components/TooltipNew/GuideTooltip.js

```diff
@@ -0,0 +1,272 @@
+          [styles[`${baseClass}__overlay--visible`]]: isVisible,
+          [styles[`${baseClass}__overlay--slide`]]: slide
+        })}
+        style={overlayBottom}
+      />
+    </React.Fragment>
+  );
+};
+
+GapOverlay.propTypes = {
+  gap: PropTypes.shape({
+    top: PropTypes.number,
+    left: PropTypes.number,
+    bottom: PropTypes.number,
+    right: PropTypes.number,
+    width: PropTypes.number,
+    height: PropTypes.number
+  }),
+  isVisible: PropTypes.bool,
+  slide: PropTypes.bool
+};
+
+class GuideTooltip extends React.PureComponent {
+  static getDerivedStateFromProps(props, state) {
+    if (props.isVisible) {
+      if (!state.lastElement) {
+        // this is the first time the tooltip is shown - don't slide from the (0, 0) origin
+        return {
+          shouldSlide: false,
+          lastElement: props.element
+        };
+      }
+
+      if (props.element !== state.lastElement) {
+        // the element has changed - slide to the next one
+        return {
+          shouldSlide: true,
+          lastElement: props.element
+        };
+      }
+
+      // window was scrolled or any other change has happened - don't slide
+      return {
+        shouldSlide: false
+      };
+    }
+
+    // reset the element when the tooltip hides
+    return {
+      lastElement: null
+    };
+  }
+
+  constructor(props) {
+    super(props);
+
+    // we don't use the following state params but we still need to re-render when the window changes
+    /* eslint-disable react/no-unused-state */
+    this.state = {
+      windowWidth: 0,
+      windowHeight: 0,
+      windowScrollX: 0,
+      windowScrollY: 0
+    };
+    /* eslint-enable react/no-unused-state */
+
+    this.handleDocumentChange = this.handleDocumentChange.bind(this);
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleDocumentChange);
+    document.addEventListener('scroll', this.handleDocumentChange);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.handleDocumentChange);
+    document.removeEventListener('scroll', this.handleDocumentChange);
+  }
+
+  handleDocumentChange() {
+    const windowWidth = window.innerWidth;
+    const windowHeight = window.innerHeight;
+    const windowScrollX = document.documentElement.scrollLeft;
+    const windowScrollY = document.documentElement.scrollTop;
+
+    /* eslint-disable react/no-unused-state */
+    this.setState({
+      windowWidth,
+      windowHeight,
+      windowScrollX,
+      windowScrollY
+    });
+    /* eslint-enable react/no-unused-state */
+  }
+
+  render() {
+    const {
+      children,
+      className,
+      placement,
+      zIndex,
+      element,
+      isVisible,
+      slide,
+      theme
+    } = this.props;
+    const referenceElement = new VirtualReference(element);
+    const rect = referenceElement.getBoundingClientRect();
+    const shouldSlide = slide && this.state.shouldSlide;
+
+    return (
+      <TooltipPortal>
```

Comment:
there is no way to pass a different `parentElementName`

Reply:
fixed